src/cryptography/hazmat/backends/openssl/x25519.py

```diff
@@ -71,5 +71,9 @@ def exchange(self, peer_public_key):
         self._backend.openssl_assert(keylen[0] > 0)
         buf = self._backend._ffi.new("unsigned char[]", keylen[0])
         res = self._backend._lib.EVP_PKEY_derive(ctx, buf, keylen)
-        self._backend.openssl_assert(res == 1)
+        if res != 1:
+            raise ValueError(
+                "Null shared key derived from public/private pair."
```

Comment:
`s/null/all zero/`?

Reply:
That doesn't really seem clearer to me (is it actually zeroes or zero bits?), but I don't feel strongly so I'll change it if you care more than I do :)